mitodl/odl-video-service
# Refactor video analytics collection

#### What are the relevant tickets?
Fixes #401 

#### What's this PR do?
- Makes all event labels == the video key
- Video playback progress is sent every 60 seconds with a `Tdddd` event (where `dddd` equals the minute with up to 3 preceding 0's).
- Records the camera angle of each event using a custom dimension (needs to be created manually in Google Analytics).
- Adds a new .env variable / django setting `GA_DIMENSION_CAMERA` that should equal the custom dimension (`"dimension1"` unless other custom dimensions already exist).
- Changes the camera angle names from `upperLeft`, `lowerRight` etc to `camera1` through `camera4`

#### How should this be manually tested?
- Set up a google analytics account if necessary
- Create a custom dimension for camera angle in GA:
  
<img width="317" alt="screen shot 2018-01-10 at 1 15 18 pm" src="https://user-images.githubusercontent.com/187676/34788369-5fe840da-f608-11e7-8b91-c63c7f9d91b0.png">


- Set `GA_TRACKING_ID` in your `.env` file to the appropriate value
- Set `GA_DIMENSION_CAMERA` in your `.env` file to `"dimension1"`
- Open a detail page for a multiangle video with the network developer tab open and start playing the video.  You should see `collect` requests sent to Google Analytics whenever you change the camera angle, play, pause, stop, etc and every time an additional minute of video is played.  The data sent with each request should include:
    - `label`: `<video.key>`
    - `cd1`: `camera<number>`
    - `ea`: `<name of event>`
- After a few minutes, log in to Google Analytics and see if the data is present (the custom dimension should be available as a `Secondary dimension`)
- Repeat some of the above with a non-multiangle video.  Everything should be about the same except that the `cd1` parameter shouldn't be sent.
